File: util/src/entities/StickerPack.ts
import { Column, Entity, JoinColumn, ManyToOne, OneToMany, OneToOne, RelationId } from "typeorm";
import { Sticker } from ".";
import { BaseClass } from "./BaseClass";

@Entity("sticker_packs")
export class StickerPack extends BaseClass {
	@Column()
	name: string;

	@Column({ nullable: true })
	description?: string;

	@Column({ nullable: true })
	banner_asset_id?: string;

	@OneToMany(() => Sticker, (sticker: Sticker) => sticker.pack, {
		cascade: true,
		orphanedRowAction: "delete",
	})
	stickers: Sticker[];

	// sku_id: string

	@Column({ nullable: true })
	@RelationId((pack: StickerPack) => pack.cover_sticker)
	cover_sticker_id?: string;

	@ManyToOne(() => Sticker, { nullable: true })
	@JoinColumn()
	cover_sticker?: Sticker;
}

File: extra/admin-api/Models/Spacebar.Models.Db/Models/StickerPack.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Spacebar.Models.Db.Models;

[Table("sticker_packs")]
public partial class StickerPack
{
    [Key]
    [Column("id", TypeName = "character varying")]
    public string Id { get; set; } = null!;

    [Column("name", TypeName = "character varying")]
    public string Name { get; set; } = null!;

    [Column("description", TypeName = "character varying")]
    public string? Description { get; set; }

    [Column("banner_asset_id", TypeName = "character varying")]
    public string? BannerAssetId { get; set; }

    [Column("cover_sticker_id", TypeName = "character varying")]
    public string? CoverStickerId { get; set; }

    [Column("coverStickerId", TypeName = "character varying")]
    public string? CoverStickerId1 { get; set; }

    [ForeignKey("CoverStickerId1")]
    [InverseProperty("StickerPacks")]
    public virtual Sticker? CoverStickerId1Navigation { get; set; }

    [InverseProperty("Pack")]
    public virtual ICollection<Sticker> Stickers { get; set; } = new List<Sticker>();
}
